lib/at_cmd_parser/src/at_cmd_decoder.c

```diff
@@ -0,0 +1,277 @@
+#include <zephyr.h>
+
+#include <at_cmd_parser.h>
+#include <at_cmd_decoder.h>
+
+#include <string.h>
+
+/* Forward declarations. */
+static void *at_CEREG_decode(struct at_param_list *param_list,
+			     u32_t valid_params);
+
+static void *at_CESQ_decode(struct at_param_list *param_list,
+			    u32_t valid_params);
+
+static void *at_NOT_CESQ_decode(struct at_param_list *param_list,
+				u32_t valid_params);
+
+static void *at_CMT_decode(struct at_param_list *param_list,
+			   u32_t valid_params);
+
+static void *at_CNUM_decode(struct at_param_list *param_list,
+			    u32_t valid_params);
+
+AT_CMD_DECODER_LIST_START(default_decoder_list)
+AT_CMD_DECODER_LIST_ENTRY("+CESQ", AT_CMD_CESQ_MODEL, at_CESQ_decode)
+AT_CMD_DECODER_LIST_ENTRY("%CESQ", AT_NOT_CESQ_MODEL, at_NOT_CESQ_decode)
+AT_CMD_DECODER_LIST_ENTRY("+CEREG:", AT_CMD_CEREG_MODEL, at_CEREG_decode)
+AT_CMD_DECODER_LIST_ENTRY("+CMT:", AT_CMD_CMT_MODEL, at_CMT_decode)
+AT_CMD_DECODER_LIST_ENTRY("+CNUM:", AT_CMD_CNUM_MODEL, at_CNUM_decode)
+AT_CMD_DECODER_LIST_END();
+
+static struct at_cmd_decoder_list const *at_cmd_decoders;
+static struct at_param_list             param_list;
+static at_cmd_decoder_handler_t         at_cmd_decoder_handler;
+
+static void *at_CMT_decode(struct at_param_list *param_list,
+			   u32_t valid_params)
+{
+	int err;
+
+	if (valid_params != 3) {
+		return NULL;
+	}
+
+	struct at_cmd_model_cmt *model =
+			k_malloc(sizeof(struct at_cmd_model_cmt));
+
+	if (!model) {
+		return NULL;
+	}
+	memset(model, 0, sizeof(struct at_cmd_model_cmt));
+
+
+	err = at_params_get_string(param_list, 0, model->sender_addr, 32);
```

Comment:
`struct at_cmd_model_cmt *model = 
                                                 k_malloc(sizeof(struct at_cmd_model_cmt));`

Because of how memory allocation works in Zephyr I though it was easier to set a large enough static size of sender_addr and instead allocate the whole struct.

https://docs.zephyrproject.org/latest/reference/kernel/memory/heap.html?highlight=memory